src/crypto.rs

```diff
@@ -0,0 +1,295 @@
+use libc::{c_char, size_t, c_void};
+use std::mem;
+use std::ptr;
+use std::slice;
+
+use crate::yaca_lib as lib;
+use crate::yaca_conv as conv;
+use crate::*;
+
+
+// TRAITS
+
+/// Trait for different contexts
+pub trait Context {
```

Comment:
Here & other pub traits. I'm disconcerted by the fact that they are implementable outside the crate. If that's unintentional, perhaps https://rust-lang.github.io/api-guidelines/future-proofing.html#c-sealed can be of use.

Reply:
Whoa, it took me a second to understand what you mean and what this document is about and how to implement for Context.

Done.